tensorflow/tsl/util/device_name_utils.cc

```diff
@@ -618,6 +618,62 @@ std::vector<string> DeviceNameUtils::GetLocalNamesForDeviceMappings(
   return OkStatus();
 }
 
+/*static*/ string DeviceNameUtils::GetDeviceNameFromStreamDeviceName(
```

Comment:
Do you want to return a `StatusOr<string>` in case of invalid streaming device name?

Reply:
Yes, now it will return `StatusOr<string>`, please see [device_name_utils.cc](https://github.com/tensorflow/tensorflow/pull/59843/commits/0951dd8cfd6afae3941c2648a37d286c03d23a2c#diff-e1cb04fa038c64b95f4eeb92945757327173c2200741f66b5ca08bc44f7b1731R644). Not only that, but I wrote a new function [`bool DeviceNameUtils::IsStreamDeviceName()`](https://github.com/tensorflow/tensorflow/pull/59843/commits/0951dd8cfd6afae3941c2648a37d286c03d23a2c#diff-e1cb04fa038c64b95f4eeb92945757327173c2200741f66b5ca08bc44f7b1731R622) to specifically validate the device name.